src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -519,13 +521,11 @@ private async Task<IEnumerable<Diagnostic>> GetProjectDiagnosticsAsync(Project p
 
                     // Have to see if this fix is still applicable.  Jump to the foreground thread
                     // to make that check.
-                    var applicable = await Task.Factory.StartNew(() =>
-                        {
-                            this.AssertIsForeground();
-                            return fix.Action.IsApplicable(document.Project.Solution.Workspace);
-                        },
-                        cancellationToken, TaskCreationOptions.None, ForegroundTaskScheduler).ConfigureAwait(false);
-                    this.AssertIsBackground();
```

Comment:
It looks like this assert is lost?

Reply:
➡️ The `await TaskScheduler.Default` guarantees we are on a thread pool thread.